### ISSUE TITLE
Metamask modal option stuck

### ISSUE THREAD

**Issue by KGuetfade** [Web3Modal/web3modal#350]
On one particular react site of mine the metamask modal option gets stuck when pressing it. When I'm not logged in yet to metamask I can press it in the modal and the app connects and everything is fine.

However whenever I am logged into metamask and there is no cached provider and I try to press the metamask option in the modal nothing happens. The modal stays there. Code is even halted and nothing happens. 

Whenever there is a cached provider and I press connect then it skips the modal and connects straight away.

So the problem happens when metamask is logged in but there is no cached provider. Metamask gets stuck when pressing it in the modal

**Comment by appleseed-iii** [comment]
bump +1

**Comment by KGuetfade** [comment]
I solved the issue

Was my fault. I was creating many instances of web3modal because of a setinterval function

**Comment by appleseed-iii** [comment]
> I solved the issue
> 
> Was my fault. I was creating many instances of web3modal because of a setinterval function

lol me too, just realized it as well

**Comment by jordanlesich** [comment]
I'm having this issue, but I've went through the app and removed any possibilities where it could be creating more than one instance of Web3Modal. The app has never held more than one Web3Modal in state, but we did instantiate a default modal before the user connects (don't remember why). 

Anyways, we removed that and the behaviour still persists at the same rate as it did before. 

Also worth noting that this only happens after deleting localStorage, closing the browser window, and it only happens about 30% of the time. We also use the SafeAppWeb3Modal from '@gnosis.pm/safe-apps-web3modal'. This doesn't seem to have any effect. 

Leaving this here as it does seem related, but I'll also make a separate issue.
